Relax absolut path checking in our 'file' scheme implementation

levitte opened the pull request:
So far, we strictly obeyed [RFC 8089], which only allows absolute paths
in a `file:` URI.  However, this seems to give a confusing user
experience, where something like `file:foo.pem` wouldn't open foo.pem,
even though it's there in the current directory, but `file:$(pwd)/foo.pem`
would.

To be less surprising for such use cases, we relax our implementation
visavi [RFC 8089] to allow relative paths.

[RFC 8089]: https://datatracker.ietf.org/doc/html/rfc8089

Fixes #27461

DDvO commented:
Nice that this generalization even simplifies the code.

petrovr commented:
This is not correction to https://github.com/openssl/openssl/issues/27461 .
Which X.509 lookup method to replace "by file" or "by dir"?

levitte commented:
"by store" is the OSSL_STORE replacement of "by file" and "by dir".

mattcaswell reviewed (dismissed):
I wonder if this is worthy of a CHANGES.md entry?

levitte commented:
> I wonder if this is worthy of a CHANGES.md entry?

Good idea.  I added it just now...  which means that you will have to re-approve as soon as I've also rebased...

openssl-machine commented:
This pull request is ready to merge

t8m commented:
Merged to the master branch. Thank you.
